agent/agent_worker.go

```diff
@@ -291,6 +294,9 @@ func (a *AgentWorker) runPingLoop(ctx context.Context, idleMonitor *IdleMonitor)
 
 	// Create the ticker
 	pingInterval := time.Second * time.Duration(a.agent.PingInterval)
+	if a.pingIntervalForTesting != 0 {
```

Comment:
I think this is fine, and would be happy to ship it. But I like to come up with suggestions. :smile:

If the only non-zero value the new field will be set to is "negligible amount of time" (1ms), and it is a proxy for 0 wait time, then maybe we should consider a bool (e.g. `noWaitBetweenPings`) and implement an actual 0 wait time. Something like:

```go
// Operations on a nil channel block forever
var testTriggerCh chan struct{}
if a.noWaitBetweenPings { 
    // Receive from a closed channel unblocks immediately, infinitely
    testTriggerCh = make(chan struct{})
    close(testTriggerCh)
}

...

for {
    select {
    case <-testTriggerCh:
        // Only happens if a.noWaitBetweenPings is true, continue below.
    ...
    }
    // and again in the jitter select
```

Reply:
Ooh interesting.

Yeah I can't immediately see any use-case for values other than basically-zero. I was _slightly vaguely_ worried that _zero wait_ would somehow behave differently to _some wait_, regarding… I dunno race conditions or something. But I'm sure it's fine, especially if it doesn't trigger your spider senses.

What do you think about the `…ForTesting` suffix on private fields that are nonsense/dangerous outside of testing?

Reply:
Sounds reasonable.

Reply:
Done!
And verified against the existing tests 👍🏼

Reply:
I like that this is orthogonal to the actual ping delay, interacting only via the `select { … }` which is exactly where such interactions belong 👌🏼